app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,54 @@
 
 class PostsController < ApplicationController
-  before_action :set_post, except: [:index]
+  before_action :set_post, except: [:index, :new, :archive]
+  before_action :authenticate_user!, except: [:index, :show]
+
+
   def index
-    @pagy, @posts = pagy(Post.all.includes(:user, :reacters), items: 10)
+    @pagy, @posts = pagy(Post.where(status: :published).includes(:user, :reacters), items: 10)
   end
 
+  def new
+    @post = Post.create(status: :draft, user_id: current_user.id)
+    redirect_to edit_post_path(@post)
+  end
+
+  def edit
+    render layout: "new_post" 
```

Comment:
layout name should be something more abstract. For example: `without_header`, `simple_layout`,...